src/main/java/ch/njol/skript/expressions/ExprRecipeCookingTime.java

```diff
@@ -0,0 +1,100 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.RecipeUtils.RegisterRecipeEvent;
+import ch.njol.skript.util.RecipeUtils.RegisterRecipeEvent.CookingRecipeEvent;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.CookingRecipe;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Recipe Cooking Time")
+@Description("The cooking time of a Cooking, Blasting, Furnace, Campfire and Smoking Recipe.")
+@Examples({
+	"register a new blasting recipe with the key \"my_recipe\":",
+		"\tset the recipe input item to raw gold named \"Impure Gold\"",
+		"\tset the recipe cooking time to 10 seconds",
+		"\tset the recipe result to gold ingot named \"Pure Gold\""
+})
+@Since("INSERT VERSION")
+public class ExprRecipeCookingTime extends PropertyExpression<Recipe, Timespan> {
+
+	static {
+		Skript.registerExpression(ExprRecipeCookingTime.class, Timespan.class, ExpressionType.PROPERTY,
+			"[the] recipe cook[ing] time [of %-recipes%]");
```

Comment:
why doesnt this use `#register`

Reply:
Because `%recipe%` needs to be null in order to be used in the SecRegisterRecipe. Same with recipe group